### PR TITLE
GenQuery2 tests for various GenQuery1 / iquest issues (main)

### PR DESCRIPTION
**Pull request by korydraughn**
All tests in `test_iquery.py` passed at the bench.

They also passed in the testing environment.

### CODE REVIEW

File: scripts/irods/test/test_iquery.py [hunk 1]
```diff
@@ -63,3 +63,20 @@ def test_iquery_returns_error_when_closing_single_quote_is_missing__issue_6393(s
...
+        attr_name = 'originalVersionId'
+        attr_value = 'ignored'
+
+        try:
+            self.user.assert_icommand(['imeta', 'add', '-C', self.user.session_collection, attr_name, attr_value])
+
+            for op in ['IN', 'in']:
+                with self.subTest(f'op={op}'):
+                    query_string = f"select META_COLL_ATTR_NAME, META_COLL_ATTR_VALUE where META_COLL_ATTR_NAME {op} ('{attr_name}')"
+                    json_string = json.dumps([[attr_name, attr_value]], separators=(',', ':'))
```
Comment (alanking) [inline scripts/irods/test/test_iquery.py:78]: This could go outside the loop. Feel free to ignore.

File: scripts/irods/test/test_iquery.py [hunk 2]
```diff
@@ -106,3 +106,13 @@ def test_iquery_supports_embedded_single_quotes__issue_5727(self):
             query_string = f"select DATA_NAME, COLL_NAME where COLL_NAME = '{self.user.session_collection}' or DATA_NAME = '{escaped}'"
             with self.subTest(f'escaped_string={escaped}'):
                 self.user.assert_icommand(['iquery', query_string], 'STDOUT', [json_string])
+
+    def test_iquery_returns_error_on_invalid_query_string__issue_5734(self):
+        query_string = "select DATA_ID, DATA_REPL_NUM, COLL_NAME, DATA_NAME, DATA_RESC_HIER, DATA_PATH, META_DATA_ATTR_VALUE " + \
+            "where META_DATA_ATTR_NAME = 'id_foo and META_DATA_ATTR_VALUE in ('260,'3261852','3261856','3261901','3080907','3083125'," + \
```
Comment (alanking) [inline scripts/irods/test/test_iquery.py:112]: For what it's worth...

This appears to be similar/the same problem as #6393 (the problem is that there is a missing single quote after `'260`). Should we combine these tests OR use the other example test from https://github.com/irods/irods/issues/5734#issuecomment-876397457?

**Review by alanking — COMMENTED** [review]
Looks good. My comments are sort of nitpicky/"efficiency" things, so feel free to ignore.

File: scripts/irods/test/test_iquery.py (hunk 1 above)
Comment (korydraughn) [reply scripts/irods/test/test_iquery.py:78]: Missed that one. Will move.

Comment (korydraughn) [reply scripts/irods/test/test_iquery.py]: done

File: scripts/irods/test/test_iquery.py (hunk 2 above)
Comment (korydraughn) [reply scripts/irods/test/test_iquery.py:114]: done

**Review by alanking — APPROVED** [review]
Squash n #.

**Comment by korydraughn** [comment]
#'d

**Pull request merged**